datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -713,6 +713,36 @@ macro_rules! get_data_page_statistics {
                 )),
                 Some(DataType::Float32) => Ok(Arc::new(Float32Array::from_iter([<$stat_type_prefix Float32DataPageStatsIterator>]::new($iterator).flatten()))),
                 Some(DataType::Float64) => Ok(Arc::new(Float64Array::from_iter([<$stat_type_prefix Float64DataPageStatsIterator>]::new($iterator).flatten()))),
+                Some(DataType::Timestamp(unit, timezone)) => {
+                    let iter = [<$stat_type_prefix Int64DataPageStatsIterator>]::new($iterator).flatten();
+                    
+                    Ok(match unit {
+                        TimeUnit::Second => {
+                            Arc::new(match timezone {
+                                Some(tz) => TimestampSecondArray::from_iter(iter).with_timezone(tz.clone()),
```

Comment:
Not tested, but perhaps [`with_timezone_opt`](https://docs.rs/arrow/latest/arrow/array/struct.PrimitiveArray.html#method.with_timezone_opt) could reduce some the repetitiveness of the code.

Reply:
that is a great suggestion